server/src/main/scala/drt/server/feeds/mag/MagFeed.scala

```diff
@@ -0,0 +1,224 @@
+    Actual = ma.arrival.actual.map(str => SDate(str).millisSinceEpoch),
+    EstimatedChox = ma.onBlockTime.estimated.map(str => SDate(str).millisSinceEpoch),
+    ActualChox = ma.onBlockTime.actual.map(str => SDate(str).millisSinceEpoch),
+    Gate = ma.gate.map(_.name.replace("Gate ", "")),
+    Stand = ma.stand.flatMap(_.name.map(_.replace("Stand ", ""))),
+    MaxPax = ma.passenger.maximum,
+    ActPax = ma.passenger.count,
+    TranPax = ma.passenger.transferCount,
+    RunwayID = None,
+    BaggageReclaimId = None,
+    FlightID = None,
+    AirportID = ma.arrivalAirport.iata,
+    Terminal = ma.arrival.terminal.getOrElse(""),
+    rawICAO = icao(ma),
+    rawIATA = iata(ma),
+    Origin = ma.departureAirport.iata,
+    Scheduled = SDate(ma.arrival.scheduled).millisSinceEpoch,
+    PcpTime = None,
+    FeedSources = Set(LiveFeedSource)
+  )
+
+  case class IataIcao(iata: String, icao: String)
+
+  case class FlightNumber(airlineCode: String, trackNumber: Option[String])
+
+  case class Gate(name: String, number: String)
+
+  case class Stand(name: Option[String],
+                   number: Option[String],
+                   provisional: Boolean,
+                   provisionalName: Option[String],
+                   provisionalNumber: Option[String])
+
+  case class BaggageClaim(name: String, number: String, firstBagReclaim: String, lastBagReclaim: String)
+
+  case class Terminal(name: String, short_name: String, number: String)
+
+  case class Passenger(count: Option[Int],
+                       maximum: Option[Int],
+                       transferCount: Option[Int],
+                       prmCount: Option[Int])
+
+  case class Timings(scheduled: String, estimated: Option[String], actual: Option[String])
+
+  case class ArrivalDetails(airport: IataIcao,
+                            scheduled: String,
+                            estimated: Option[String],
+                            actual: Option[String],
+                            terminal: Option[String],
+                            gate: Option[String])
+
+  object JsonSupport extends SprayJsonSupport with DefaultJsonProtocol {
+    implicit val PassengerFormat: RootJsonFormat[Passenger] = jsonFormat4(Passenger)
+    implicit val TerminalFormat: RootJsonFormat[Terminal] = jsonFormat3(Terminal)
+    implicit val BaggageClaimFormat: RootJsonFormat[BaggageClaim] = jsonFormat4(BaggageClaim)
+    implicit val StandFormat: RootJsonFormat[Stand] = jsonFormat5(Stand)
+    implicit val GateFormat: RootJsonFormat[Gate] = jsonFormat2(Gate)
+    implicit val FlightNumberFormat: RootJsonFormat[FlightNumber] = jsonFormat2(FlightNumber)
+    implicit val TimingsFormat: RootJsonFormat[Timings] = jsonFormat3(Timings)
+    implicit val iataIcaoFormat: RootJsonFormat[IataIcao] = jsonFormat2(IataIcao)
+    implicit val ArrivalDetailsFormat: RootJsonFormat[ArrivalDetails] = jsonFormat6(ArrivalDetails)
+    implicit val magArrivalFormat: RootJsonFormat[MagArrival] = jsonFormat16(MagArrival)
```

Comment:
and here.

Reply:
Sorted. Not sure how I got them into that state!